app/assets/javascripts/vocab_sheet.js

```diff
@@ -91,51 +72,97 @@ $(document).ready(function() {
     }
 
     function checkForMaxLength(elem) {
-      notes = elem.val();
+      var notes = elem.val();
+      var maxLength = elem.attr('maxlength');
 
       if (notes.length >= maxLength) {
         elem.addClass('max-length-reached');
 
         setTimeout(function() {
-          textBox.removeClass('max-length-reached');
+          textBoxes.removeClass('max-length-reached');
         }, 1000);
       }
     }
 
     function setCharacterCount(elem) {
-      clearTimeout(typeTimer);
-
-      typeTimer = setTimeout(function() {
-        notes = elem.val();
+      var notes = elem.val();
+      var maxLength = elem.attr('maxlength');
 
-        elem.siblings('.character-count__wrap')
-          .children('.character-count__count')
-          .text(500 - notes.length);
+      elem.siblings('.character-count__wrap')
+        .children('.character-count__count')
+        .text(maxLength - notes.length);
 
-        updateNotes(formAction, signId, notes);
-      }, 100);
-    }
-
-    function updateNotes(action, signId, notes) {
-      $.ajax({
-        url: action,
-        method: 'PUT',
-        data: {
-          sign_id: signId,
-          notes: notes,
-        },
-        headers: {
-          'X-CSRF-Token': $('meta[name="authenticity-token"]').attr('content'),
-        },
-      }).done(function(data) {
-        // console.log("success!", data)
-      }).fail(function(error) {
-        console.error(error.statusText);
-      });
+      action = getFormAction(elem);
+      signId = getSignIdFromAction(action);
     }
 
     checkCharacterCount();
   };
 
+  if ($('.vocab-sheet__text-input').length > 0) {
+    $('.vocab-sheet__text-input').keyup(function() {
+      var field = $(this);
+      action = getFormAction(field);
+      signId = getSignIdFromAction(action);
+      var data = {signId};
+
+      if (field.hasClass('item-name')) {
```

Comment:
I think all these cases can be single line ifs?
if (field.hasClass('item-name')) { data['name'] = field.val(); }
if (field.hasClass('item-maori-name')) { data['maoriName'] = field.val(); }
if (field.hasClass('item-notes')) { data['notes'] = field.val();}